Failure changing number_of_nodes count on aws_redshift_cluster resource

It appears that the `number_of_nodes` value for a `aws_redshift_cluster` can't be changed once created.

### Terraform Version
0.6.14

### Affected Resource(s)
- aws_redshift_cluster

### Terraform Configuration Files
```hcl
resource "aws_redshift_cluster" "default" {
  cluster_identifier           = "test"
  database_name                = "test"
  master_username              = "test"
  master_password              = "testtest123"
  node_type                    = "dc1.large"
  number_of_nodes              = "2"
  cluster_subnet_group_name    = "${aws_redshift_subnet_group.private.id}"
  vpc_security_group_ids       = ["${aws_vpc.default.default_security_group_id}"]
  publicly_accessible          = "false"
  preferred_maintenance_window = "sun:00:00-sun:02:00"
  encrypted                    = "true"
}
```

### Expected Behavior
Redshift cluster size updated from 2 to 3

### Actual Behavior
```
Error applying plan:

1 error(s) occurred:

* aws_redshift_cluster.default: [WARN] Error modifying Redshift Cluster (test): InvalidParameterCombination: No modifications were requested
        status code: 400, request id: 973b43d7-076a-11e6-8145-a7ce291aa817
```

### Steps to Reproduce
1. Update `number_of_nodes`from `2` to `3`
1. Do a `terraform apply`

Hi @bsiegel 

Was able to reproduce the failure easily with a new test - will implement the new functionality now and then PR it. Quick question though - what happens if the user changes both the number of nodes && and node_type (i.e. from dc1.large to dc1.xlarge)

Will it resize the entire cluster?

```
make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSRedshiftCluster_updateNodeCount' 2>~/tf.log
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /vendor/)
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSRedshiftCluster_updateNodeCount -timeout 120m
=== RUN   TestAccAWSRedshiftCluster_updateNodeCount
--- FAIL: TestAccAWSRedshiftCluster_updateNodeCount (602.85s)
	testing.go:172: Step 1 error: Error applying: 1 error(s) occurred:

		* aws_redshift_cluster.default: [WARN] Error modifying Redshift Cluster (tf-redshift-cluster-3599417419286307578): InvalidParameterCombination: No modifications were requested
			status code: 400, request id: aef5f463-0827-11e6-bf7b-3b003af9e5ca
FAIL
exit status 1
FAIL	github.com/hashicorp/terraform/builtin/providers/aws	602.870s
```

--- a/builtin/providers/aws/resource_aws_redshift_cluster.go
+++ b/builtin/providers/aws/resource_aws_redshift_cluster.go
@@ -375,6 +375,7 @@ func resourceAwsRedshiftClusterUpdate(d *schema.ResourceData, meta interface{}) 
 		} else {
 			req.ClusterType = aws.String("single-node")
 		}
+		req.NodeType = aws.String(d.Get("node_type").(string))
 	}
 
 	if d.HasChange("cluster_security_groups") {

--- a/builtin/providers/aws/resource_aws_redshift_cluster_test.go
+++ b/builtin/providers/aws/resource_aws_redshift_cluster_test.go
@@ -71,6 +71,39 @@ func TestAccAWSRedshiftCluster_publiclyAccessible(t *testing.T) {
 	})
 }
 
+func TestAccAWSRedshiftCluster_updateNodeCount(t *testing.T) {
+	var v redshift.Cluster
+
+	ri := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+	preConfig := fmt.Sprintf(testAccAWSRedshiftClusterConfig_basic, ri)
+	postConfig := fmt.Sprintf(testAccAWSRedshiftClusterConfig_updateNodeCount, ri)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSRedshiftClusterDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: preConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSRedshiftClusterExists("aws_redshift_cluster.default", &v),
+					resource.TestCheckResourceAttr(
+						"aws_redshift_cluster.default", "number_of_nodes", "1"),
+				),
+			},
+
+			resource.TestStep{
+				Config: postConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSRedshiftClusterExists("aws_redshift_cluster.default", &v),
+					resource.TestCheckResourceAttr(
+						"aws_redshift_cluster.default", "number_of_nodes", "2"),
+				),
+			},
+		},
+	})
+}
+
 func testAccCheckAWSRedshiftClusterDestroy(s *terraform.State) error {
 	for _, rs := range s.RootModule().Resources {
 		if rs.Type != "aws_redshift_cluster" {
@@ -272,6 +305,24 @@ func TestResourceAWSRedshiftClusterMasterUsernameValidation(t *testing.T) {
 	}
 }
 
+var testAccAWSRedshiftClusterConfig_updateNodeCount = `
+provider "aws" {
+	region = "us-west-2"
+}
+
+resource "aws_redshift_cluster" "default" {
+  cluster_identifier = "tf-redshift-cluster-%d"
+  availability_zone = "us-west-2a"
+  database_name = "mydb"
+  master_username = "foo_test"
+  master_password = "Mustbe8characters"
+  node_type = "dc1.large"
+  automated_snapshot_retention_period = 0
+  allow_version_upgrade = false
+  number_of_nodes = 2
+}
+`
+
 var testAccAWSRedshiftClusterConfig_basic = `
 provider "aws" {
 	region = "us-west-2"
@@ -284,7 +335,7 @@ resource "aws_redshift_cluster" "default" {
   master_username = "foo_test"
   master_password = "Mustbe8characters"
   node_type = "dc1.large"
-  automated_snapshot_retention_period = 7
+  automated_snapshot_retention_period = 0
   allow_version_upgrade = false
 }`
 
@@ -344,7 +395,7 @@ resource "aws_redshift_cluster" "default" {
   master_username = "foo"
   master_password = "Mustbe8characters"
   node_type = "dc1.large"
-  automated_snapshot_retention_period = 7
+  automated_snapshot_retention_period = 0
   allow_version_upgrade = false
   cluster_subnet_group_name = "${aws_redshift_subnet_group.foo.name}"
   publicly_accessible = false
@@ -406,7 +457,7 @@ resource "aws_redshift_cluster" "default" {
   master_username = "foo"
   master_password = "Mustbe8characters"
   node_type = "dc1.large"
-  automated_snapshot_retention_period = 7
+  automated_snapshot_retention_period = 0
   allow_version_upgrade = false
   cluster_subnet_group_name = "${aws_redshift_subnet_group.foo.name}"
   publicly_accessible = true

